be/src/exprs/agg/window_funnel.h

```diff
@@ -33,12 +33,38 @@ struct ComparePairFirst final {
 
 enum FunnelMode : int { DEDUPLICATION = 1, FIXED = 2, DEDUPLICATION_FIXED = 3 };
 
+namespace InteralTypeOfFunnel {
+template <PrimitiveType primitive_type>
+struct TypeTraits {};
+
+template <>
+struct TypeTraits<TYPE_INT> {
```

Comment:
What's the purpose? What's the difference of `RunTimeTypeTraits` ?

Reply:
> What's the purpose? What's the difference of `RunTimeTypeTraits` ?

For DATE and DATETIME we process internal type of CppType(DateValue and TimestampValue), for BIGINT and INT just CppType, so we should extract concrete type first.